### PR TITLE
8325464: GCCause.java out of sync with gcCause.hpp

### PR DESCRIPTION
**Pull request by leveretconey**
These two files (`GCCause.java` and `gcCause.hpp`) should be in sync by design, see comments in these two files. However, some recent changes (e.g. [JDK-8240239](https://bugs.openjdk.org/browse/JDK-8240239)) to `gcCause.hpp` were not simultaneously reflected in `GCCause.java`. This patch updates `GCCause.java` to keep sync with latest `gcCause.hpp`.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8325464](https://bugs.openjdk.org/browse/JDK-8325464): GCCause.java out of sync with gcCause.hpp (**Bug** - P4)

### Reviewers
 * [Stefan Karlsson](https://openjdk.org/census#stefank) (@stefank - **Reviewer**)
 * [Thomas Schatzl](https://openjdk.org/census#tschatzl) (@tschatzl - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/17766/head:pull/17766` \
`$ git checkout pull/17766`

Update a local copy of the PR: \
`$ git checkout pull/17766` \
`$ git pull https://git.openjdk.org/jdk.git pull/17766/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 17766`

View PR using the GUI difftool: \
`$ git pr show -t 17766`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/17766.diff">https://git.openjdk.org/jdk/pull/17766.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/17766#issuecomment-1933432138)

### CODE REVIEW

File: src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/gc/shared/GCCause.java [hunk 1]
```diff
@@ -60,11 +62,11 @@ public enum GCCause {
   _z_allocation_rate ("Allocation Rate"),
   _z_allocation_stall ("Allocation Stall"),
   _z_proactive ("Proactive"),
+  _z_high_usage ("High Usage"),
```
Comment (stefank) [inline src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/gc/shared/GCCause.java:65]: gcCause.hpp has the Shenandoah and ZGC blocks swapped. Could you fix it in this change as well?

**Review by stefank — CHANGES_REQUESTED** [review]

File: src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/gc/shared/GCCause.java (hunk 1 above)
Comment (leveretconey) [reply src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/gc/shared/GCCause.java:70]: OK, I have swapped the Shenandoah and ZGC blocks here.

**Review by stefank — APPROVED** [review]
Looks good.

**Review by tschatzl — APPROVED** [review]
Lgtm.

**Pull request closed without merging**